### ISSUE TITLE
Linker error when linking against a package with cmm-sources in profiling mode

### ISSUE THREAD

**Issue by TerrorJack** [haskell/cabal#7153]
**Describe the bug**
https://github.com/TerrorJack/cmm-prof-bug-repro is a minimal repro package with an executable `foo` and a library with a cmm file. When building it in profiling mode, linker error occurs.

**To Reproduce**

```
$ cabal v2-run foo --enable-profiling
```

Results in the following error message:

```
/home/terrorjack/workspace/cmm-prof-bug-repro/dist-newstyle/build/x86_64-linux/ghc-8.8.4/foo-0/x/foo/build/foo/foo-tmp/Main.p_o:ghc_3.c:function Main_main1_info: error: undefined reference to 'succ'
/home/terrorjack/workspace/cmm-prof-bug-repro/dist-newstyle/build/x86_64-linux/ghc-8.8.4/foo-0/build/libHSfoo-0-inplace_p.a(Foo.p_o):ghc_3.c:function foozm0zminplace_Foo_czusucc_info: error: undefined reference to 'succ'
/home/terrorjack/workspace/cmm-prof-bug-repro/dist-newstyle/build/x86_64-linux/ghc-8.8.4/foo-0/build/libHSfoo-0-inplace_p.a(Foo.p_o):ghc_3.c:function foozm0zminplace_Foo_mySucc_info: error: undefined reference to 'succ'
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
```

Without `--enable-profiling` it runs fine.

**Expected behavior**
It should build `foo` and run it, which outputs `234` just like the non-profiling build.

**System information**
 - Operating system: Linux x64
 - `cabal`, `ghc` versions: `cabal-install-3.2.0.0`, `ghc-8.8.4`

**Comment by bgamari** [comment]
It's possible that this is fixed by #7183 .

**Comment by alt-romes** [comment]
This has probably been fixed by 296a515fbd9a58c13c6096ca25dab46a2e4cbf6e, and I cannot reproduce with 3.10 (which includes that commit.

Feel free to re-open if it is not fixed for you.
